inc/gimage.h

```diff
@@ -155,18 +155,20 @@ extern Color GDrawColorBrighten(Color col, int by);
 extern int GImageWriteGImage(GImage *gi, char *filename);
 extern int GImageWrite_Bmp(GImage *gi, FILE *fp);
 extern int GImageWriteBmp(GImage *gi, char *filename);
-extern GImage *GImageRead_Bmp(FILE *file);
+extern GImage *GImageRead_Bmp(FILE *file, int *success);
```

Comment:
As far as I can see, all the specific image readers used to signal success/failure by non-NULL or NULL return value. Is there any benefit from explicitly setting `success` flag?

Reply:
I guess we could change `success` to a boolean (along with its name). My concern was that the previous version of the code wound up printing error messages without the filename whereas the code before this PR would include it. Assuming this rough arrangement we need some way of telling the called function whether to print its own message or not.